### PR TITLE
Expunge Expr.ufl_domain Part II

### PR DESCRIPTION
**Pull request by pbrubeck**
# Description
This was still triggering warnings at interpolation (and when trying to output non-Lagrange functions).

### CODE REVIEW

**Review by connorjward — DISMISSED** [review]
Looks sensible.

**Comment by JDBetteridge** [comment]
Hurrah, now we are only raising <checks notes> 13414 warnings on CI!

This _is a massive improvement_ over the 20947 warnings we generated previously.

@pbrubeck do you fancy opening a similar PR to expunge this warning from UFL itself?

**Comment by pbrubeck** [comment]
> Hurrah, now we are only raising 13414 warnings on CI!
> 
> This _is a massive improvement_ over the 20947 warnings we generated previously.
> 
> @pbrubeck do you fancy opening a similar PR to expunge this warning from UFL itself?

I am not entirely sure we are completely eliminating this warnigns. Maybe I'll first open a ufl PR that fails instead of raising the warning to detect where else we need to change our code, and then open another PR to remove the warning.

**Comment by wence-** [comment]
> > Hurrah, now we are only raising 13414 warnings on CI!
> > This _is a massive improvement_ over the 20947 warnings we generated previously.
> > @pbrubeck do you fancy opening a similar PR to expunge this warning from UFL itself?
> 
> I am not entirely sure we are completely eliminating these warnigns. Maybe I'll first open a ufl PR that fails instead of raising the warning to detect where else we need to change our code, and then open another PR to remove the warning.

you can either turn warnings into errors in pytest, or, if you run just a python script, do `python -Werror --pdb script.py` to turn warnings into errors and drop into the debugger: https://docs.python.org/3/using/cmdline.html#cmdoption-W

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -79,6 +79,8 @@ jobs:
             --install icepack \
             --install irksome \
             --install femlium \
+            --package-branch ufl pbrubeck/ufl_domain_failure \
+            --package-branch tsfc pbrubeck/expunge-ufl-domain \
```
Comment (pbrubeck) [inline .github/workflows/build.yml:83]:
```suggestion
```

**Comment by pbrubeck** [comment]
We went down to 1249 warnings after making sure this issue is never triggered by the tests.

**Review by connorjward — DISMISSED** [review]
Thanks for taking the time to do this.

File: firedrake/preconditioners/patch.py [hunk 2]
```diff
@@ -778,7 +779,8 @@ def initialize(self, obj):
             J = ctx.Jp or ctx.J
             bcs = ctx._problem.bcs
 
-        mesh = J.ufl_domain()
+        V, _ = map(operator.methodcaller("function_space"), J.arguments())
```
Comment (connorjward) [inline firedrake/preconditioners/patch.py:782]:
```suggestion
        V = J.arguments()[0].function_space()
```

**Review by connorjward — CHANGES_REQUESTED** [review]
Just clearing up a strange bit of code. Otherwise fine.

**Review by dham — DISMISSED** [review]

**Review by ReubenHill — APPROVED** [review]

**Pull request merged**